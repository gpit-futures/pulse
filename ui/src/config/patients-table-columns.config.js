import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
