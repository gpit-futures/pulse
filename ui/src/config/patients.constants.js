import { getAgeYears } from '../utils/time-helpers.utils';

export const AGE_RANGE_PREFIX = 'ageRange';
export const DEPARTMENT_PREFIX = 'department';

export const patientsDepartmentPredicate = departmentName => patient => patient.department === departmentName;

export const patientsAgeRangesPredicate = (from, to) => patient => from <= getAgeYears(patient.dateOfBirth) && getAgeYears(patient.dateOfBirth) <= to;

export const patientsDepartments = [
  { name: 'Community Care', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Community Care') },
  { name: 'Hospital', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Hospital') },
  { name: 'Mental Health', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Mental Health') },
  { name: 'Neighbourhood', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Neighbourhood') },
  { name: 'Primary Care', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Primary Care') },
];

export const patientsAgeRanges = [
  { name: '19-30', from: 19, to: 30, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(19, 30) },
  { name: '31-60', from: 31, to: 60, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(31, 60) },
  { name: '61-80', from: 61, to: 80, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(61, 80) },
  { name: '>80', from: 81, to: 999, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(81, 999) },
];

const ageRangeForClinicalSearch = [{ name: '11-18', from: 11, to: 18, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(11, 18) }];

export const patientAgeRangesForClinicalSearch = ageRangeForClinicalSearch.concat(patientsAgeRanges);