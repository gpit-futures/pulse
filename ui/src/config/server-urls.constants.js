export const apiUrls = {
  INITIALISE: '/api/initialise',
  LOGOUT: '/api/logout',
};

export const usersUrls = {
  PATIENTS_URL: '/api/patients',
  USER_ACCOUNT_URL: '/api/user',
  BASIC_PATIENT_SEARCH: '/api/search/patient/table',
  ADVANCED_PATIENT_SEARCH: '/api/patients/advancedSearch',
  PROFILE_APP_PREFERENCES: '/api/application',
  LIST_ORDERS: '/api/terminology/list/order',
  PICTURES: '/api/pictures',
  CLINICAL_QUERY_SEARCH: '/api/patients/querySearch',
  FEEDS: '/api/feeds',
};
