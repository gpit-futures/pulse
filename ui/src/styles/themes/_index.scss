@import 'gray';
@import 'purple';
@import 'leedsPHR';
