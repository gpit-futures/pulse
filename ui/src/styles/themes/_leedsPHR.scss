$dodger-blue: #3596f4;
$lucky-point: #18276d;
$white: #ffffff;

$base-color: $dodger-blue;
$auxiliary-color: $lucky-point;

$table-color: lighten($base-color, 40%);;
$table-text: #5c5c5c;
$colored-table: (
  bg: #ffffff,
  color: $table-text,
  border-color: #d4d2d2,
  border-width: 1px,

  header-bg: #e0e0e0,
  header-color: #333,
  header-highlighted-bg: darken(#e0e0e0, 10%),
  header-highlighted-color: #333,

  row-bg: #ffffff,
  row-odd-bg: #ffffff,
  row-hover-bg: $auxiliary-color,
  row-odd-hover-bg: $auxiliary-color,
  row-color: $table-text,
  row-odd-color: $table-text,
  row-hover-color: #fff,
  row-odd-hover-color: #fff,

  row-sorted-bg: lighten(#e0e0e0, 7%),
  row-sorted-odd-bg: lighten(#e0e0e0, 7%),
  row-sorted-hover-bg: $auxiliary-color,
  row-sorted-odd-hover-bg: $auxiliary-color,
  row-sorted-color: $table-text,
  row-sorted-odd-color: $table-text,
  row-sorted-hover-color: #fff,
  row-sorted-odd-hover-color: #fff,

  row-highlighted-bg: #b4b4b4,
  row-highlighted-color: #ffffff,
);

$text-color: #303030;
$text-on-base-color: #ffffff;

$colored-buttons: (
    default: #ffffff,
    primary: #337AB7,
    success: #017A41,
    info: #31b0d5,
    warning: #F0AD4E,
    danger: #DA534F,
    theme: $auxiliary-color
);


/**
  * COLORS for base configuration
  */
$body-bg: #eee;

/**
  * COLORS for header
  */

$header-bg: $white;
$header-btn: (
    bg: $dodger-blue,
    color: $text-on-base-color,
    bg-hover: $lucky-point,
    color-hover: $text-on-base-color,
    bg-active: $lucky-point,
    color-active: $text-on-base-color
);
$header-toolbar: (
    bg: $lucky-point,
    color: $text-on-base-color,
);

/**
  * COLORS for search
  */
  $search-btn: (
      bg: $base-color,
      color: $text-on-base-color,
      bg-hover: $auxiliary-color,
      color-hover: $text-on-base-color
  );
  $search-btn-toggle: (
      bg: $dodger-blue,
      color: $text-on-base-color,
      bg-hover: $auxiliary-color,
      color-hover: $text-on-base-color,
      border: $base-color,
      border-hover: $auxiliary-color
  );

/**
  * COLORS for footer
  */
  $footer: (
    bg: $white,
    color: #5c5c5c
  );

/**
  * COLORS for dashboard
  */
  $board: (
    border: #c9c8c8,
    radius: 4px,
    header-bg: $base-color,
    header-color: $text-on-base-color,
    preview-ratio: 60%,
    item-bg: $white,
    item-odd-bg: $white,
    item-color: $text-color,
    item-bg-hover: $auxiliary-color,
    item-color-hover: $white,
  );

/**
  * COLORS for panels
  */
  $panel-border-color: #d4d2d2;
  $panel-heading-bg: $base-color;
  $panel-heading-color: $text-on-base-color;
  $panel-control-bg: transparent;
  $panels: (
    panel-default: (
      bg: #fff,
      heading-bg: $base-color,
      heading-color: $text-on-base-color,
      border-color: $base-color,
      border-width: 2px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-primary: (
      bg: transparent,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 3px,
      control-bg: transparent
    ),
    panel-secondary: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-subaccordion: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-small: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    )
  );

/**
  * COLORS for forms
  */
  $forms: (
    bg: #fff,
    color: #555555,
    disabled-bg: #ebebeb,
    disabled-color: #5c5c5c,
    readonly-bg: #fff,
    readonly-color: #555,

    border-color: #c5c4c4,
    border-error-color: #a94442,
    border-success-color: #2b542c,
    border-width: 1px,
    border-radius: 4px,
    addon-color: #222,
    label-color: #333,
    label-control-color: #333,
    label-required-color: #a32d44,
    control-static-color: #333,
  );

body.themeLeedsPHR {
  @import 'theme-source/index';
  @import 'theme-leedsPHR/index';
}