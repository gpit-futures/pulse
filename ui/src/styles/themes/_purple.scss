$purple: #59056e;
$purple-light: #6a1e7c;

$base-color: $purple;

$auxiliary-color: lighten($base-color, 20%);

$table-color: lighten($base-color, 40%);;
$table-text: #333;
$colored-table: (
  bg: #F3F3F3,
  color: $table-text,
  border-color: #B4B5B4,
  border-width: 1px,

  header-bg: lighten(saturate(adjust-hue($table-color, -10.5516), 9.6510), 6.2745),
  header-color: $table-text,
  header-highlighted-bg: $table-color,
  header-highlighted-color: $table-text,

  row-bg: #F3F3F3,
  row-odd-bg: #FDFDFD,
  row-hover-bg: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
  row-odd-hover-bg: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
  row-color: $table-text,
  row-odd-color: $table-text,
  row-hover-color: $table-text,
  row-odd-hover-color: $table-text,

  row-sorted-bg: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
  row-sorted-odd-bg: lighten(saturate(adjust-hue($table-color, -9.8551), 8.9921), 24.1176),
  row-sorted-hover-bg: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
  row-sorted-odd-hover-bg: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
  row-sorted-color: $table-text,
  row-sorted-odd-color: $table-text,
  row-sorted-hover-color: $table-text,
  row-sorted-odd-hover-color: $table-text,

  row-highlighted-bg: #b4b4b4,
  row-highlighted-color: #ffffff,
);

$text-color: #303030;
$text-on-base-color: #ffffff;

$colored-buttons: (
  default: #ffffff,
  primary: #337AB7,
  success: lighten(saturate(adjust-hue($base-color, 9.3005), 22.4138), 1.1765),
  info: #31b0d5,
  warning: #F0AD4E,
  danger: #DA534F,
  theme: $base-color
);

/**
  * COLORS for base configuration
  */
  $body-bg: #fff;

/**
  * COLORS for header
  */

  $header-bg: $purple;
  $header-btn: (
    bg: $purple,
    color: $text-on-base-color,
    bg-hover: $purple-light,
    color-hover: $text-on-base-color,
    bg-active: $purple-light,
    color-active: $text-on-base-color
  );
  $header-toolbar: (
    bg: $purple-light,
    color: $text-on-base-color,
  );

/**
  * COLORS for search
  */
  $search-btn: (
    bg: #000000,
    color: #ffffff,
    bg-hover: #000000,
    color-hover: #ffffff
  );
  $search-btn-toggle: (
    bg: #000000,
    color: #ffffff,
    bg-hover: #000000,
    color-hover: #ffffff,
    border: #000000,
    border-hover: #000000
  );

/**
  * COLORS for footer
  */
  $footer: (
    bg: #eee,
    color: #757584
  );

/**
  * COLORS for dashboard
  */
  $board: (
    border: #B4B5B4,
    radius: 0,
    header-bg: lighten(saturate(adjust-hue($table-color, -10.5516), 9.6510), 6.2745),
    header-color: #333,
    preview-ratio: 60%,
    item-bg: #F3F3F3,
    item-odd-bg: #FDFDFD,
    item-color: #333,
    item-bg-hover: lighten(desaturate(adjust-hue($table-color, -9.8551), 7.9490), 20.9804),
    item-color-hover: #333,
  );

/**
  * COLORS for panels
  */
  $panel-border-color: #D8D8D8;
  $panel-heading-bg: #E8E8E8;
  $panel-control-bg: #F3F3F3;
  $panel-heading-color: $text-color;
  $panels: (
    panel-default: (
      bg: #fff,
      heading-bg: $base-color,
      heading-color: $text-on-base-color,
      border-color: $base-color,
      border-width: 2px,
      border-radius: 0,
      control-bg: $panel-control-bg
    ),
    panel-primary: (
      bg: transparent,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 0,
      control-bg: transparent
    ),
    panel-secondary: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 0,
      control-bg: $panel-control-bg
    ),
    panel-subaccordion: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 0,
      control-bg: $panel-control-bg
    ),
    panel-small: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 0,
      control-bg: $panel-control-bg
    )
  );

/**
  * COLORS for forms
  */
  $forms: (
    bg: #fff,
    color: #555555,
    disabled-bg: #eeeeee,
    disabled-color: #555555,
    readonly-bg: #fff,
    readonly-color: #555555,

    border-color: #c3c3c3,
    border-error-color: #a94442,
    border-success-color: #2b542c,
    border-width: 1px,
    border-radius: 2px,
    addon-color: #303030,
    label-color: #a9a9a9,
    label-control-color: #949494,
    label-required-color: #a32d44,
    control-static-color: #000000,
  );

body.themePurple {
  @import 'theme-source/index';
}