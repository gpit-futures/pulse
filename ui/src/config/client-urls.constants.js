import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
