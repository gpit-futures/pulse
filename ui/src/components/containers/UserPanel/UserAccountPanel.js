import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { lifecycle } from 'recompose';
import packageJSON from '../../../../package.json';

import PTButton from '../../ui-elements/PTButton/PTButton';
import UserAccountPanelSettings from './UserAccountPanelSettings';
import userAccountSelector from './selectors';
import { initialiseSelector } from '../App/selectors';
import userImage from '../../../assets/images/user.jpg'
import { clientUrls } from '../../../config/client-urls.constants';
import { unmountOnBlur } from '../../../utils/HOCs/unmount-on-blur.utils';
import { logoutStart } from '../../../ducks/logout.duck';
import { themeConfigs } from '../../../themes.config';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ logoutStart }, dispatch) });

@connect(initialiseSelector)
@connect(userAccountSelector, mapDispatchToProps)
@lifecycle(unmountOnBlur)
export default class UserAccountPanel extends PureComponent {
  static propTypes = {
    user: PropTypes.shape().isRequired,
    actions: PropTypes.objectOf(PropTypes.func).isRequired,
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  render() {
    const { user, actions, onClick, onClose, initialiseData } = this.props;
    const varsionOfPulseTile = packageJSON.version;
    const varsionOfReact = packageJSON.dependencies.react;

    return (
      <div className="dropdown-user dropdown-menu-right dropdown-menu">
        <div
          className="user-profile-image"
          onClick={() =>{
            this.context.router.history.push(clientUrls.USER_PROFILE);
            onClick('')
          }}
        >
          <div className="img">
            <img src={userImage} alt="" />
          </div>
        </div>
        <div className="user-profile-info">
          <div
            className="user-profile-info__item name"
            onClick={() =>{
              this.context.router.history.push(clientUrls.USER_PROFILE);
              onClick('')
            }}
          >{user.given_name} {user.family_name}</div>
          <div className="user-profile-info__descr">
            <div className="user-profile-info__item role">{user.role}</div>
            <div className="user-profile-info__item email">{user.email}</div>
            <div className="user-profile-info__item birthday">10/05/2099</div>
          </div>
          <div className="specification">
            <div className="user-profile-info__title registered">Registered GP</div>
            <div className="user-profile-info__item version-back">About Showcase Stack; PulseTile version {varsionOfPulseTile}/QEWD_Ripple version {initialiseData.version}</div>
            <div className="user-profile-info__item version-front">React version {varsionOfReact}</div>
          </div>
          <PTButton className="btn btn-theme btn-block btn-signout">
            <div onClick={actions.logoutStart}>
              <span className="brn-text">Sign Out</span> <i className="btn-icon fa fa-sign-out" />
            </div>
          </PTButton>
        </div>
        { themeConfigs.isShowUserProfileSettings && <UserAccountPanelSettings onClose={onClose} /> }
      </div>
    )
  }
}
