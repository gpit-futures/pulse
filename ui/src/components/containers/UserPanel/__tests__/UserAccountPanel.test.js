import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import UserAccountPanel from '../UserAccountPanel';
import { themeConfigs } from '../../../../themes.config';

Enzyme.configure({ adapter: new Adapter() })

const mockStore = configureStore();
const storeResource = {
  userAccount: {
    sub: '28AD8576-1948-4C84-8B5E-55FB7EE027CE',
    given_name: 'Bob',
    family_name: 'Smith',
    email: 'bob.smith@gmail.com',
    tenant: null,
    role: 'IDCR',
    roles: [
      'IDCR'
    ]
  },
  patientsInfo: {
    browserTitle: 'PulseTile',
    logoB64: 'testLogo',
    themeColor: 'green',
    title: 'PulseTile',
  },
  initialiseData: {
    token: 'd4281221-3950-4019-b207-18e49086dab8',
    mode: 'demo',
    version: '2.32.1',
  },
};
const context = {
  router: {
    history: {
      test: 'test',
      goBack: () => {},
      push: () => {},
    },
  },
};
const onClick = () => console.log('test')
const storeProfilePage = mockStore(Object.assign({
  router: {
    location: {
      pathname: '/',
      search: '',
      hash: '#/profile',
    },
  } }, storeResource));
describe('Component <UserAccountPanel />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(
      <UserAccountPanel
        store={storeProfilePage}
        onClick={onClick}
      />, { context }).dive().dive().dive();
    expect(component).toMatchSnapshot();
    component.setContext(context);
    component.find('.user-profile-image').simulate('click');
    component.find('.name').simulate('click');
  });

  it('should renders with props correctly with isShowUserProfileSettings ', () => {
    themeConfigs.isShowUserProfileSettings = true;
    const component = shallow(
      <UserAccountPanel
        store={storeProfilePage}
        onClick={onClick}
      />, { context }).dive().dive().dive();
    component.setContext(context);
    expect(component.find('Connect(UserAccountPanelSettings)')).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
