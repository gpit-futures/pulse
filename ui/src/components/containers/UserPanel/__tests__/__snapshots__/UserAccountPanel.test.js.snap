// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <UserAccountPanel /> should renders with props correctly 1`] = `
<div
  className="dropdown-user dropdown-menu-right dropdown-menu"
>
  <div
    className="user-profile-image"
    onClick={[Function]}
  >
    <div
      className="img"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
  <div
    className="user-profile-info"
  >
    <div
      className="user-profile-info__item name"
      onClick={[Function]}
    >
      Bob
       
      Smith
    </div>
    <div
      className="user-profile-info__descr"
    >
      <div
        className="user-profile-info__item role"
      >
        IDCR
      </div>
      <div
        className="user-profile-info__item email"
      >
        bob.smith@gmail.com
      </div>
      <div
        className="user-profile-info__item birthday"
      >
        10/05/2099
      </div>
    </div>
    <div
      className="specification"
    >
      <div
        className="user-profile-info__title registered"
      >
        Registered GP
      </div>
      <div
        className="user-profile-info__item version-back"
      >
        About Showcase Stack; PulseTile version 
        1.0.0
        /QEWD_Ripple version 
        2.32.1
      </div>
      <div
        className="user-profile-info__item version-front"
      >
        React version 
        15.6.2
      </div>
    </div>
    <PTButton
      className="btn btn-theme btn-block btn-signout"
    >
      <div
        onClick={[Function]}
      >
        <span
          className="brn-text"
        >
          Sign Out
        </span>
         
        <i
          className="btn-icon fa fa-sign-out"
        />
      </div>
    </PTButton>
  </div>
</div>
`;

exports[`Component <UserAccountPanel /> should renders with props correctly with isShowUserProfileSettings  1`] = `
<div
  className="dropdown-user dropdown-menu-right dropdown-menu"
>
  <div
    className="user-profile-image"
    onClick={[Function]}
  >
    <div
      className="img"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
  <div
    className="user-profile-info"
  >
    <div
      className="user-profile-info__item name"
      onClick={[Function]}
    >
      Bob
       
      Smith
    </div>
    <div
      className="user-profile-info__descr"
    >
      <div
        className="user-profile-info__item role"
      >
        IDCR
      </div>
      <div
        className="user-profile-info__item email"
      >
        bob.smith@gmail.com
      </div>
      <div
        className="user-profile-info__item birthday"
      >
        10/05/2099
      </div>
    </div>
    <div
      className="specification"
    >
      <div
        className="user-profile-info__title registered"
      >
        Registered GP
      </div>
      <div
        className="user-profile-info__item version-back"
      >
        About Showcase Stack; PulseTile version 
        1.0.0
        /QEWD_Ripple version 
        2.32.1
      </div>
      <div
        className="user-profile-info__item version-front"
      >
        React version 
        15.6.2
      </div>
    </div>
    <PTButton
      className="btn btn-theme btn-block btn-signout"
    >
      <div
        onClick={[Function]}
      >
        <span
          className="brn-text"
        >
          Sign Out
        </span>
         
        <i
          className="btn-icon fa fa-sign-out"
        />
      </div>
    </PTButton>
  </div>
  <Connect(UserAccountPanelSettings) />
</div>
`;
